ner_telhub/model/data_models.py

```diff
@@ -0,0 +1,292 @@
+import csv
+import numpy as np
+from typing import Iterable, List, Any, Tuple
+
+from PyQt6.QtCore import (
+    QAbstractTableModel, Qt, 
+    QReadWriteLock, QModelIndex,
+    QReadLocker, QWriteLocker,
+    pyqtBoundSignal, pyqtSignal,
+    QObject, QDateTime
+)
+
+from ner_processing.master_mapping import DATA_IDS
+from ner_processing.data import Data
+from ner_telhub.utils.threads import Worker
+
+
+class DataModel(QAbstractTableModel):
```

Comment:
What is the use of QAbstractTableModel? I've never used it before and it sounds cool

Reply:
Same thing as QAbstractListModel but it stores multiple columns of data rather than a one-dimensional list. It's good for our data as we have two columns, one for time and the other for an actual data value.